app/src/main/res/layout/fragment_ai_prompt.xml

```diff
@@ -92,22 +91,47 @@
                     android:layout_height="wrap_content"
                     android:gravity="top|start"
                     android:hint="@string/aiPromptPlaceholder"
-                    android:inputType="textMultiLine|textCapSentences|textAutoCorrect"
-                    android:theme="@style/AiCursorAndPrimaryColorTheme" />
+                    android:inputType="textMultiLine|textCapSentences|textAutoCorrect" />
 
             </com.google.android.material.textfield.TextInputLayout>
 
-            <com.google.android.material.button.MaterialButton
-                android:id="@+id/generateButton"
-                style="@style/TextButtonPrimaryMedium"
+            <FrameLayout
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
-                android:layout_gravity="end"
-                android:enabled="false"
-                android:text="@string/aiPromptValidateButton"
-                android:theme="@style/AiCursorAndPrimaryColorTheme"
-                tools:enabled="true" />
+                android:layout_gravity="end">
 
+                <com.google.android.material.button.MaterialButton
+                    android:id="@+id/generateButton"
+                    style="@style/TextButtonPrimaryMedium"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:enabled="false"
+                    android:text="@string/aiPromptValidateButton"
+                    tools:enabled="true" />
+
+                <LinearLayout
+                    android:id="@+id/generationLoader"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center_vertical"
+                    android:visibility="gone">
+
+                    <TextView
+                        style="@style/BodyMedium.Tertiary"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="center_vertical"
```

Comment:
Is this gravity really useful ? As it's already declare in the parent
(It doesn't seem in my IDE)

Reply:
This one helps align the text and the spinner inside the LinearLayout, which has nothing to do with the one already declared in the parent. With a normal text size the TextView is always bigger than the spinner and only the spinner needs to be centered vertically, but I imagine that if you downscale the text size of your phone then the spinner could be bigger than the text and then you'd need to center vertically the text instead